### PR TITLE
feat: Cosmos SDK v47 integration (backport: #1782)

### PR DESCRIPTION
**Pull request by mmsqe**
for more [info](https://github.com/evmos/ethermint/pull/1782)

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: rpc/backend/utils.go [hunk 1]
```diff
@@ -318,12 +318,12 @@ func TxLogsFromEvents(events []abci.Event, msgIndex int) ([]*ethtypes.Log, error
 func ParseTxLogsFromEvent(event abci.Event) ([]*ethtypes.Log, error) {
 	logs := make([]*evmtypes.Log, 0, len(event.Attributes))
 	for _, attr := range event.Attributes {
-		if !bytes.Equal(attr.Key, []byte(evmtypes.AttributeKeyTxLog)) {
+		if !bytes.Equal([]byte(attr.Key), []byte(evmtypes.AttributeKeyTxLog)) {
```
Comment (yihuang) [inline rpc/backend/utils.go]:
```suggestion
		if !attr.Key != evmtypes.AttributeKeyTxLog {
```

File: rpc/types/utils.go [hunk 2]
```diff
@@ -235,8 +241,8 @@ func BaseFeeFromEvents(events []abci.Event) *big.Int {
 		}
 
 		for _, attr := range event.Attributes {
-			if bytes.Equal(attr.Key, []byte(feemarkettypes.AttributeKeyBaseFee)) {
-				result, success := new(big.Int).SetString(string(attr.Value), 10)
+			if bytes.Equal([]byte(attr.Key), []byte(feemarkettypes.AttributeKeyBaseFee)) {
```
Comment (yihuang) [inline rpc/types/utils.go]:
```suggestion
			if attr.Key == feemarkettypes.AttributeKeyBaseFee {
```

**Review by yihuang — APPROVED** [review]
LGTM, only minor suggestions.

File: go.mod [hunk 3]
```diff
@@ -177,46 +184,46 @@ require (
...
-	github.com/gin-gonic/gin => github.com/gin-gonic/gin v1.9.0
-	// use cosmos flavored protobufs
-	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
+	github.com/gin-gonic/gin => github.com/gin-gonic/gin v1.7.0
 	github.com/syndtr/goleveldb => github.com/syndtr/goleveldb v1.0.1-0.20210819022825-2ae1ddf74ef7
-	// use cometbft
-	github.com/tendermint/tendermint => github.com/cometbft/cometbft v0.34.28
-	// https://github.com/crypto-org-chain/tm-db/tree/release/v0.6.x
-	github.com/tendermint/tm-db => github.com/crypto-org-chain/tm-db v0.6.8-0.20230424032152-87c7e7f4fb61
+	google.golang.org/grpc => github.com/mmsqe/grpc-go v0.0.0-20230726140020-924503e66efe
```
Comment (mmsqe) [inline go.mod]: there's bug [fix](https://github.com/grpc/grpc-go/pull/6306/files) in newer grpc-go, we can't rely on old err msg [check](https://github.com/crypto-org-chain/ethermint/blob/20253cee104adc8a92803aaa7518079f389db404/tests/integration_tests/test_grpc_only.py#L106)

**Pull request merged**